mtkclient: init at 2.0.1-unstable-2025-09-26

timschumi opened the pull request:
[mtkclient](https://github.com/bkerler/mtkclient) is a low-level tool to interface with Mediatek-based devices while in brom or preloader mode.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

timschumi committed:
mtkclient: init at 2.0.1-unstable-2025-09-26

Co-authored-by: Dom Rodriguez <shymega@shymega.org.uk>

yzhou216 commented on pkgs/by-name/mt/mtkclient/package.nix:
```diff
@@ -0,0 +1,43 @@
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication rec {
```
```suggestion
python3Packages.buildPythonApplication {
```

yzhou216 reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

timschumi commented on pkgs/by-name/mt/mtkclient/package.nix:
```diff
@@ -0,0 +1,43 @@
+    colorama
+    fusepy
+    hatchling
+    mock
```
Bumped the package to the latest revision now, `mock` has been replaced with `unittest.mock`, which is built-in.

timschumi commented:
Pulled in udev files from #447819 in latest push. `50-android.rules` is due to be obsolete with systemd 258, so I'll drop that once #427968 lands in `master`, and `51-edl.rules` probably doesn't do anything at all because it's for qcom.

@shymega I stole the authorship information from 6814394537d60587b6d57250fe693275faf63f36, let me know if you want something else listed there (or nothing at all).

timschumi commented:
Also updated the version number to `2.0.1-unstable-...`, I think that's what most can agree on and is most in line with the guidelines.

timschumi commented:
> You mean the meta attrset? You didn't specify a line range in the link, so I wasn't sure.

What I'm talking about is the `Co-authored-by` line in the commit message. I copied your name and email from the authorship information of the referenced commit.

timschumi commented:
Added the proper `sourceProvenance` definitions in the latest push and dropped the udev rules files in the push before that.

Continuing a remaining point of discussion from #447819 here:

> maybe we should keep the udev rules until this version of systemd has been in unstable, 25.05, and the upcoming 25.11 for a few weeks?

Any branch that doesn't have the new systemd version still has `android-udev-rules` available, which is a more updated version of the `50-android.rules` in `mtkclient`.

shymega commented:
> Added the proper `sourceProvenance` definitions in the latest push and dropped the udev rules files in the push before that.

Nice!

 
> Continuing a remaining point of discussion from #447819 here:
> 
> > maybe we should keep the udev rules until this version of systemd has been in unstable, 25.05, and the upcoming 25.11 for a few weeks?
> 
> Any branch that doesn't have the new systemd version still has `android-udev-rules` available, which is a more updated version of the `50-android.rules` in `mtkclient`.

Sounds like a good plan, thanks for looking into this.

I'm going to go ahead and approve this PR as its looking good!

shymega reviewed (approved):
LGTM, and beat me to the punch!
